mlx/ops.cpp

```diff
@@ -696,6 +696,41 @@ split(const array& a, int num_splits, StreamOrDevice s /* = {} */) {
   return split(a, num_splits, 0, to_stream(s));
 }
 
+std::vector<array> meshgrid(
+    const std::vector<array>& arrays,
+    bool sparse /* = false */,
+    std::string_view indexing /* = "xy" */,
```

Comment:
Let's use a `std::string` here solely for the sake of MLX C (to save binding a new type).

Reply:
It might work with `string_view` (@andresy )?

Reply:
@abeleinin do you mind using `std::string` here?

Reply:
At @abeleinin we should use a `std::string` here, do you mind updating it?